Sources/CryptoSwift/Salsa20.swift

```diff
@@ -0,0 +1,262 @@
+        
+        
+        // 11 - 14
+        for i in 0..<4 {
+            let start = addPos + (i*4)
+            
+            let bytes = key[start..<(start + 4)]
+            ctx.input[i + 11] = wordNumber(bytes)
+        }
+        
+        // iv
+        ctx.input[6] = wordNumber(iv[0..<4])
+        ctx.input[7] = wordNumber(iv[4..<8])
+        ctx.input[8] = 0
+        ctx.input[9] = 0
+        
+        
+        return ctx
+    }
+    
+    func encryptBytes(message:[UInt8]) throws -> [UInt8] {
+        
+        guard let ctx = context else {
+            throw Error.MissingContext
+        }
+        
+        guard var keyStream = keyStream else {
+            throw Error.MissingContext
+        }
+        
+        var c:[UInt8] = [UInt8](count: message.count, repeatedValue: 0)
+        
+        let bytes = message.count
+        
+        while (true) {
+            for i in 0..<bytes {
+                c[i] = message[i] ^ keyStream[index]
+                //reset index every 64 byte, regenerate keyStream
+                index = (index + 1) & 0x3F
+                if index == 0 {
+                    ctx.input[8] = ctx.input[8] &+ 1
+                    if (ctx.input[8] == 0) {
+                        ctx.input[9] = ctx.input[9] &+ 1
+                        /* stopping at 2^70 bytes per nonce is user's responsibility */
+                        if (ctx.input[9] == 0) {
+                            throw Error.LimitOfNonceExceeded
+                        }
+                    }
+                    keyStream = wordToByte(ctx.input)!
+                    self.keyStream = keyStream
+                }
+            }
+            
+            return c
+        }
+    }
+    
+    private final func quarterround(b:UInt32, _ c:UInt32, _ d:UInt32) -> UInt32 {
+        return rotateLeft(b &+ c, d)
+    }
+    
+    /**
+     Reset keystream to start again
+     */
+    public func reset() {
+        //reset index counter
+        index = 0
+        
+        //reset position
+        context?.input[8] = 0
+        context?.input[9] = 0
+        
+        //reset stream key
+        self.keyStream = wordToByte((context?.input)!)!
+    }
+}
+
+// MARK: - Cipher
+
+extension Salsa20: Cipher {
+    public func cipherEncrypt(bytes:[UInt8]) throws -> [UInt8] {
+        return try self.encrypt(bytes)
+    }
+    
+    public func cipherDecrypt(bytes: [UInt8]) throws -> [UInt8] {
+        return try self.decrypt(bytes)
+    }
+}
+
+// MARK: Helpers
+
+/// Change array to number. It's here because arrayOfBytes is too slow
+func wordNumber(bytes:ArraySlice<UInt8>) -> UInt32 {
+    var value:UInt32 = 0
+    for i:UInt32 in 0..<4 {
+        let j = bytes.startIndex + Int(i)
+        value = value | UInt32(bytes[j]) << (8 * i)
+    }
+    
+    return value
+}
+
+func littleEndian(data: NSData, range: Range<Int>) -> UInt32 {
+    var val: UInt32 = 0
+    data.getBytes(&val, range: NSRange(range))
```

Comment:
this introduce Foundation dependency here where it should not be required. There is problem with Foundation for Linux so in general code that relies on Foundation is separated and optional.